src/services/upload/readFileService.ts

```diff
@@ -0,0 +1,104 @@
+import { FILE_TYPE } from 'services/fileService';
+import { ENCRYPTION_CHUNK_SIZE } from 'types';
+import { logError } from 'utils/sentry';
+import { MIN_STREAM_FILE_SIZE } from './uploadService';
+
+const TYPE_VIDEO = 'video';
+const TYPE_HEIC = 'HEIC';
+export const TYPE_IMAGE = 'image';
+const EDITED_FILE_SUFFIX = '-edited';
+
+export async function getFileData(reader: FileReader, file: globalThis.File) {
+    return file.size > MIN_STREAM_FILE_SIZE
+        ? getFileStream(reader, file)
+        : await getUint8ArrayView(reader, file);
+}
+
+export function getFileType(receivedFile: globalThis.File) {
+    let fileType: FILE_TYPE;
+    switch (receivedFile.type.split('/')[0]) {
+        case TYPE_IMAGE:
+            fileType = FILE_TYPE.IMAGE;
+            break;
+        case TYPE_VIDEO:
+            fileType = FILE_TYPE.VIDEO;
+            break;
+        default:
+            fileType = FILE_TYPE.OTHERS;
+    }
+    if (
+        fileType === FILE_TYPE.OTHERS &&
+        receivedFile.type.length === 0 &&
+        receivedFile.name.endsWith(TYPE_HEIC)
+    ) {
+        fileType = FILE_TYPE.IMAGE;
+    }
+    return fileType;
+}
+
+export function getFileOriginalName(file: globalThis.File) {
+    let originalName: string = null;
+
+    const isEditedFile = file.name.endsWith(EDITED_FILE_SUFFIX);
+    if (isEditedFile) {
+        originalName = file.name.slice(0, -1 * EDITED_FILE_SUFFIX.length);
+    } else {
+        originalName = file.name;
+    }
+    return originalName;
+}
+
+function getFileStream(reader: FileReader, file: globalThis.File) {
```

Comment:
nit: We can it more generic and pass the `chunkSize` as a input to `getFileStream` and `fileChunkReaderMaker` method. This two func doesn't need to be aware of `ENCRYPTION_CHUNK_SIZE`.